medusa/post_processor.py

```diff
@@ -287,6 +288,16 @@ def list_associated_files(self, file_path, base_name_only=False, subtitles_only=
 
         return file_path_list
 
+    @staticmethod
+    def rar_basename(filepath, filelist):
+        videofile = os.path.basename(filepath)
+        rars = (x for x in filelist if rarfile.is_rarfile(x))
+
+        for rar in rars:
+            content = rarfile.RarFile(rar).namelist()
```

Comment:
rar_file = [os.path.basename(f).rpartition('.')[0].lower() for f in file_list if helpers.get_extension(f).lower() == 'rar']